usecases/scenario_testrun_usecase.go

```diff
@@ -97,7 +97,41 @@ func (usecase *ScenarioTestRunUsecase) ActivateScenarioTestRun(ctx context.Conte
 			if err != nil {
 				return models.ScenarioTestRun{}, err
 			}
+			result.ScenarioId = scenario.Id
 			return *result, err
 		},
 	)
 }
+
+func (usecase *ScenarioTestRunUsecase) ListTestRunByScenarioId(ctx context.Context,
+	organizationId, scenarioId string,
+) ([]models.ScenarioTestRun, error) {
+	if err := usecase.enforceSecurity.ListTestRuns(organizationId); err != nil {
+		return nil, err
+	}
+	testruns, err := usecase.repository.ListTestRunsByScenarioID(ctx,
+		usecase.executorFactory.NewExecutor(), scenarioId)
+	if err != nil {
+		return nil, err
+	}
+	if len(testruns) > 0 {
+		for _, testrun := range testruns {
+			testrun.ScenarioId = scenarioId
+		}
+	}
+	return testruns, nil
+}
+
+func (usecase *ScenarioTestRunUsecase) GetTestRunById(ctx context.Context,
+	testRunId string,
+) (models.ScenarioTestRun, error) {
+	testrun, err := usecase.repository.GetTestRunByID(ctx,
+		usecase.executorFactory.NewExecutor(), testRunId)
+	if err != nil {
+		return models.ScenarioTestRun{}, err
+	}
+	if testrun == nil {
+		return models.ScenarioTestRun{}, nil
+	}
+	return *testrun, nil
```

Comment:
here (and on every test run returned by the "list" method), let's run an `enforceSecurity.ReadTestRun(testRun)` (method to create) before returning them to the API (similarly to what is done in other usecases, see users for instance)